Initial Stab at Privacy Considerations + Sentence to Abstract about did:earth as Cosmos focused IIDs (#13)
EarthProgram/did-earth: README.md

A DID that uses this method MUST begin with the following prefix `did:earth`. The prefix string MUST be in lowercase. The remainder of the DID after the prefix, is specified below:

#### Method Specific Identifier
The `did:earth` method-specific identifier (`method-specific-id`) is made up of a `version`, `chainspace` and a `namespace` component.

> The `version` is defined as a number that identifies a specific version of the `did:earth` method specification (e.g. 1, 2, 3 etc.). The version number enables future enhancements to the specification defining `did:earth` identifiers.

Reviewer: ```suggestion
The `version` is defined as an integer that identifies a specific version of `did:earth` method operations such as 
create, read, update, and deactivate. The version number increases by one for each breaking change in the specification, enabling future enhancements to the specification while retaining backwards compatibility for long-lived identifiers. 

Implementers SHOULD maintain compatibility with all existing versions likely to be in use. For application chains, this means supporting the version that was current when the chain first adopted did:earth functionality, as well as all subsequent versions if possible. DID resolvers--which may be off-chain--will likely need to maintain support for all versions until convinced there are no outstanding DIDs using that version. Software which is asked to handled a version # that it does not support MUST return an error.
```

The `chainspace` is defined as a string that identifies a specific Cosmos blockchain (e.g. "ixo", "regen", "cosmos") where the DID reference is stored.

> The `namespace` is defined as an alphanumeric string that identifies a specific Cosmos module in the `chainspace` (e.g. "nft", "bank", "staking") where the DID reference is stored.

Reviewer: ```suggestion
The `namespace` is an alphanumeric string that identifies a distinct namespace managed by the application chain's name server module. These namespaces, e.g., "nft", "bank", "staking", identify where the on-chain asset is maintained on that particular chain. It is used to route incoming resolution requests to the correct asset module.
```

The `did:earth` method support offline creation of DIDs.
An offline generated `did:earth` DID must be unique by having the `unique-id` component be derived from the initial public key of the DID. 

> For an `Ed25519` public key, the first 16 bytes of the base-58 representation of the 256-bit public key is used to generate the unique-id.
> multibase(multicodec(public_key))

#### did:earth method syntax
```abnf

> earth-did          = "did:earth:" version ":" chainspace ":" namespace ":" unique-id

Reviewer: I think Shaun prefers the version to be after the human-friendly parts.
```suggestion
earth-did          = "did:earth:" chainspace ":" namespace ":" version ":"  unique-id
```

Resources may be provided in-line or by secure reference. Inline resources are appropriate only for use cases that need to directly include the resource in the IID Document. In many cases, this is a privacy problem. However, for some use cases, resources must be specified for on-chain execution, which justifies the added bytes and potential disclosure risk. The resource descriptor provides for a flexible representation of various mime types, compression, and encoding, as required for the use.

This approach allows token owners to manage privacy in three key ways:

1.  Avoids posting potentially sensitive information on-chain in an unavoidably public and irrevocable manner.
2.  Provides a service endpoint that can apply appropriate privacy and security checks before revealing information.

> 3.  The hashgraph resource descriptor type obscures not only the content of the linked resource, but also the quantity of resource objects.

Author: @jandrieu do you want me to put this paragraph back here then? All I did was refactor this out into a privacy considerations section to start with.

Reviewer: Nope. This can go.

### Privacy Considerations

> The `did:earth` method is a fully compliant IID and DID. As such the privacy considerations taken into account are applicable to this method. The reader is directed towards the privacy considerations discussed in [section 10 of the W3C DID Core specification](https://w3c.github.io/did-core/#privacy-considerations).

Reviewer: Unfortunately, this paragraph doesn't say much.

What we should highlight are the following points
1. IIDs, e.g., `did:earth:ixo:nft:1:abc` are designed to represent on-chain assets, as such the identifier itself is never used to refer to real-world objects like people, with associated privacy requirements.
2. IID references, e.g., `did:earth:ixo:nft:1:abc#creator` may in fact be used to refer to real people and organizations. As such, care must be taken to ensure that any associated personal data be managed off chain with appropriate privacy mechanisms such as the ability to remove the data from public disclosure.
4. IID references specified as Linked Resources offer several privacy-agile ways to associate the on-chain asset with arbitrary resources, providing a way for `did:earth:ixo:nft:1:abc#creator` to refer to an off-chain data store with the sensitive data. For example, an NFT's creator could be specified by a link and a hash to a Verifiable Credential (or a simple JSON file) that states the creator's name. Storing that name on-chain would create a regulatory problem. Linked Resources allow the information to be shared while ensuring the ability to honor requests for deletion.

example json of a linked resource specifying the creator
```json
"linkedResource" : {
  "id" : "#creator",
  "path": "/creator",
  "rel":"dc:creator",
  "type" : "iid:ResourceDescriptor",
  "proof" : [{
     "type": "hash",
     "stage": "encoded",
     "value" : "bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi"}],
   "resourceFormat" : "application/did+ld+json",
   "compression" : "none",
   "endpoint": "https://earthprogram.directory?listing=did:earth:ixo:nft:1:abc123#owner"
}
```
The VC retrieved from that endpoint contains the personal data describing the author. This personal information is never stored on-chain, rather, it is retrievable from a mutable data store, which retains the ability to honor GDPR deletion requests.

Author: I do agree with this. Yours is much better. But I still think there is value in directing the reader to the DID Core spec privacy considerations. Or is it just implicit that we expect the implementor to know about those?

#### Linked Resources

Linked resources within IIDs support privacy in three key three key ways:

1.  Avoids posting potentially sensitive information on-chain in an unavoidably public and irrevocable manner.
2.  Provides a service endpoint that can apply appropriate privacy and security checks before revealing information.
3.  The hashgraph resource descriptor type obscures not only the content of the linked resource, but also the quantity of resource objects.

> 

Reviewer: We can replace this whole section with something that captures the points in my previous comment. We probably don't want literal bullet points, although an introductory set of bullet titles followed by a paragraph for each bullet should work well.

> > Note: Not sure where this JSON goes. Feels hanging?

Reviewer: That should be in the linked resources definition section, rather than the privacy considerations section.

Author: Actually, I think we can just remove this. The current linked resources section contains this proof object within the larger linkedResource json:

```javascript
{
	"linkedResource": [{

		"path"(optional): // IID Resource path for this resource in the asset namespace, e.g., /myResource.png

			"id"(optional): // IID Reference ID for this resource, e.g., #myResource.png

			"rel"(optional): // the relationship of this resource to the IID Asset

			"type": "nft:ResourceDescriptor", // The JSON-LD type of this resource

		"proof": [ // an array of proofs, any of which may be used
			{
				"type"(optional): "hash" | "hashgraph" | "hashset" // the form of proof used to verify the resource as authentic
				"stage"(optional): "raw" | "compressed" | "encrypted" | "encoded" // the 
				"value"(optional): hash | hashgraph | hashset, // the actual proof for this particular resource
			}
		],
		"resourceFormat": media type, // the IANA media-type of the linked resource

		"compression"(optional): "gzip" | "none", // the compression (performed on the raw resource) of an inline resource

		"encryption": [open ended
			for arbitrary extensibility
		], // the encryption technique applied after compression and before encoding

		"encoding"(optional): "native" | "multibase" | "string", // the encoding (after compression and encryption) of an inline resource; "native" means the inline resource is native JSON-LD with neither compression nor encryption

		"endpoint"(optional): url, // a URL at which this resource can be retrieved before decrypting and decrompressing

		"resource"(optional): "..."
		a representation of the resource
		for inline
		distribution,
		encoded as per "encoding"
    }]
}
```

Anyreason to include the proof json on its own aswell?
